tests/snmp/test_snmp_link_local.py

```diff
@@ -0,0 +1,62 @@
+import pytest
+from tests.common.helpers.snmp_helpers import get_snmp_facts
+from tests.common import config_reload
+
+pytestmark = [
+    pytest.mark.topology('t0', 't1', 't2', 'm0', 'mx'),
+    pytest.mark.device_type('vs')
+]
+
+
+@pytest.fixture(autouse=True, scope='module')
+def config_reload_after_test(duthosts,
+                             enum_rand_one_per_hwsku_frontend_hostname):
+    yield
+    duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
+    config_reload(duthost, config_source='config_db')
+
+
+@pytest.mark.bsl
+def test_snmp_link_local_ip(duthosts,
+                            enum_rand_one_per_hwsku_frontend_hostname,
+                            nbrhosts, tbinfo, localhost, creds_all_duts):
+    """
+    Test SNMP query to DUT over link local IP
+      - Send SNMP query over link local IP from one of the BGP Neighbors
+      - Get SysDescr from snmpfacts
+      - compare result from snmp query over link local IP and snmpfacts
+    """
+    duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
+    hostip = duthost.host.options['inventory_manager'].get_host(
+        duthost.hostname).vars['ansible_host']
+    snmp_facts = get_snmp_facts(
+        localhost, host=hostip, version="v2c",
+        community=creds_all_duts[duthost.hostname]["snmp_rocommunity"],
+        wait=True)['ansible_facts']
+    # Get link local IP of mamangement interface
+    ip_cmd = 'ip addr show eth0 | grep "inet6" | grep "link"\
+              | awk "{print $2}" | cut -d/ -f1'
+    link_local_ips = duthost.shell(ip_cmd)['stdout_lines']
+    # configure link local IP in config_db
+    for ip in link_local_ips:
+        if ip.split()[1].lower().startswith('fe80'):
+            link_local_ip = ip.split()[1]
+            break
+    # configure link local IP in config_db
+    duthost.shell(
+            'redis-cli -n 4 hset "MGMT_INTERFACE|eth0|{}" "gwaddr" "fe80::1"'
+            .format(link_local_ip))
+    # Restart snmp service to regenerate snmpd.conf with
+    # link local IP configured in MGMT_INTERFACE
+    duthost.shell("systemctl restart snmp")
+    stdout_lines = duthost.shell("docker exec snmp snmpget \
+                                 -v2c -c {} {}%eth0 1.3.6.1.2.1.1.1.0"
```

Comment:
>1.3.6.1.2.1.1.1.0 [](http://example.com/codeflow?start=52&length=17)

Better to have a named constant, so audience could understand what is the OID meaning.

Reply:
updated